spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/integration/test/DockerComposeFactory.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.integration.test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+
+import com.palantir.docker.compose.DockerComposeExtension;
+import com.palantir.docker.compose.configuration.DockerComposeFiles;
+import com.palantir.docker.compose.connection.DockerMachine;
+import com.palantir.docker.compose.connection.waiting.HealthChecks;
+import org.junit.jupiter.api.extension.BeforeAllCallback;
+import org.junit.jupiter.api.extension.Extension;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.cloud.dataflow.integration.test.util.ResourceExtractor;
+import org.springframework.cloud.dataflow.integration.test.util.TestProperties;
+
+/**
+ * Factory utility that helps to install SCDF & Skipper from Docker Compose files.
+ *
+ * Test fixture applies the same docker compose files used for the Data Flow local installation:
+ *  - https://dataflow.spring.io/docs/installation/local/docker/
+ *  - https://dataflow.spring.io/docs/installation/local/docker-customize/
+ *
+ * The Palantir DockerMachine and DockerComposeExtension are used to programmatically deploy the docker-compose files.
+ *
+ * All important bootstrap parameters are configurable via the {@link TestProperties} properties and variables.
+ *
+ * @author Christian Tzolov
+ */
+public class DockerComposeFactory {
+
+	private static final Logger logger = LoggerFactory.getLogger(DockerComposeFactory.class);
+
+	/**
+	 * Data Flow version to use for the tests.
+	 */
+	public static final String DEFAULT_DATAFLOW_VERSION = "2.4.0.BUILD-SNAPSHOT";
```

Comment:
Can we infer the default property values via a property file somehow? It feels weird to update the versions at the Java file every time we move from one version to the other.

Reply:
What about if we start to tag these images per branch? Currently `latest` always points to master build on scdf/skipper. We could add additional tags, `latest-master`, `latest-24x`, etc when we eventually have maintenance branches.

Reply:
@ilayaperumalg , the DockerMachine and DockerComposeExtension had to be static (forgot the exact reason)! In result the standard Spring properties are not applicable.  Or do you mean something else? 
@jvalkeal,  not sure i understand.

Reply:
@ilayaperumalg actually you have a point. The DataFlow URL and Platform name are not used in the DockerCompose initialization but only inside the test and can be made Spring properties. I will change this

Reply:
Well I meant if we hardcode a version we could just do it for known tag?